### PR TITLE
Polish

### PR DESCRIPTION
**Pull request by anik31**
### Updates added
- Added debounce for search
- Refactor
- Error handling
- UI & bug fixes

### [Live link](https://deploy-preview-6--vacay-notes.netlify.app/)

### CODE REVIEW

**Review by sainath7878 — COMMENTED** [review]
Everything looks good. Good to merge.

File: src/hooks/useDebounce.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+    useEffect(() => {
+        const timerId = setTimeout(() => setDebouncedValue(value), delay);
+        // Cancel the timeout if value changes (also on delay change or unmount)
+        // This is how we prevent debounced value from updating if value is changed ...
+        // .. within the delay period. Timeout gets cleared and restarted.
+        return () => clearTimeout(timerId);
+      }, [value, delay] // Only re-call effect if value or delay changes
+    );
+    return debouncedValue;
```
Comment (BalajiHarinath) [inline src/hooks/useDebounce.js:14]: Appreciation: Good that you are using a custom hook for debounce.

**Review by BalajiHarinath — APPROVED** [review]
Everything looks good. You can merge.

File: src/hooks/useDebounce.js (hunk 1 above)
Comment (anik31) [reply src/hooks/useDebounce.js:14]: Thanks

**Pull request merged**